src/ffi/mod.rs

```diff
@@ -151,6 +151,27 @@ fn deserialize_features(features: *const c_char) -> Result<Features, telio_resul
     }
 }
 
+#[cfg(target_os = "android")]
+#[no_mangle]
+/// Initialize OS certificate store, should be called only once. Without call to telio_init_cert_store
+/// telio will not be able to verify https certificates in the system certificate store.
+/// # Params
+/// - `env`:    see https://developer.android.com/training/articles/perf-jni#javavm-and-jnienv
+/// - `ctx`:    see https://developer.android.com/reference/android/content/Context
+pub extern "C" fn telio_init_cert_store(env: *mut jni::sys::JNIEnv, ctx: jni::sys::jobject) {
+    static INIT: Once = Once::new();
+    INIT.call_once(|| match unsafe { jni::JNIEnv::from_raw(env) } {
+        Ok(env) => {
+            if let Err(err) = rustls_platform_verifier::android::init_hosted(&env, ctx.into()) {
```

Comment:
Do you think reporting the error with a status code or something like that makes sense?
It seems the `init_hosted()` already does the `Once` logic for you

Reply:
> Do you think reporting the error with a status code or something like that makes sense?

Good idea - I've added a fallback feature flag for hardcoded list so, when this fails application can actually try to 'recover'.

> It seems the init_hosted() already does the Once logic for you

That's an implementation detail of `init_hosted` which is not guaranteed to be there in future. On the other hand readme says:

> This only needs to be called once and the verifier will be valid for the lifetime of your app's process.

Which doesn't *require* one call but clearly states that only one call is *needed*, and I think it's better to make sure on our side that we don't make unnecessary calls.

Reply:
Yeah, makes sense to guard from multiple calls on our side :+1:

Regarding the error code, it would be nice to have it plus maybe an error for the case when this function is called more than once. However, this will complicate the `Once` logic a bit, so IDK, your call.